external/buildscripts/build_all_osx.pl

```diff
@@ -17,6 +20,10 @@
 
 my @thisScriptArgs = ();
 my @passAlongArgs = ();
+
+my $buildX64 = $buildMachine || ($hostArch != $m1Arch);	     # Build x64 conditional
+my $buildArm64  = $buildMachine || ($hostArch == $m1Arch);   # Build arm64 conditional
```

Comment:
```suggestion
my $buildArm64  = $buildMachine || ($hostArch eq $m1Arch);   # Build arm64 conditional
```

Reply:
iirc in perl string comparison doesn't use `==` and `!=` those are for integer comparison: https://metacpan.org/pod/perlop#Equality-Operators